Docs: Plugins Revamp

dericksozo opened the pull request:
(docs): Re-structuring the sidebar for plugins, renaming pages, and also re-writing the contents for the plugins explained page.

dericksozo commented:
Hey @PazYanoverr, I'm also taking care of #498 on this PR.

dericksozo commented:
The latest draft for #499 is in this PR, @PazYanoverr. Please let me know if this is alright, or I should open a separate PR going forward.

PazYanoverr commented:
@dericksozo - A different note that may be relevant here- in https://docs.amplication.com/getting-started/plugins/ when pressing [develop your own custom plugins](https://docs.amplication.com/plugins/overview/), a weird page which is not part of the documentation is opened.

dericksozo commented:
I appreciate the feedback, @PazYanoverr. I'll look into this one and push a fix.

PazYanoverr commented on docs/how-to/choose-configure-update-database.md:
```diff
@@ -0,0 +1,64 @@
+
+Amplication supports multiple database options, with PostgreSQL being the default choice. Here are the steps to choose your database:
+
+1. When [creating a new service](/first-service/) in Amplication, you'll be required to select a database.
```
This is relevant only in case creating a service using the wizard. When creating a service using Jovu- the user can ask for a different database than the default.
Let's make it clearer

PazYanoverr commented on docs/how-to/choose-configure-update-database.md:
```diff
@@ -0,0 +1,64 @@
+2. Choose from the available options, which may include:
+   - PostgreSQL (default)
+   - MySQL
+   - Microsoft SQL Server
```
Mongo is missing here

PazYanoverr commented on docs/how-to/choose-configure-update-database.md:
```diff
@@ -0,0 +1,64 @@
+
+Consider factors such as your team's familiarity with the database, scalability needs, and specific feature requirements when making your choice.
+
+![](./../getting-started/assets/first-service/database.png)
```
This is an old screenshot. Let's replace

PazYanoverr commented on docs/how-to/choose-configure-update-database.md:
```diff
@@ -0,0 +1,64 @@
+   - Username
+   - Password
+
+3. Replace the `host`, `port`, `user`, `dbName`, and `password` values with your specific database credentials. Click the Save button and your credentials will be applied by the plugin.
```
I think you need to re-generate the code to get these updates in your git provider as a PR. Let's add

PazYanoverr commented on docs/how-to/choose-configure-update-database.md:
```diff
@@ -0,0 +1,64 @@
+
+1. Proceed to the Installed Plugins tab. Uninstall your current database plugin.
+2. Click on the Database plugin category in the left sidebar and install a new database plugin.
+3. When you're ready to apply the changes click on the `Generate the code` button in the right sidebar.
```
Let's add a "summary" sentence - that's it, the generated code will include all the changes required in order to use the new database instead of the old one.
Something like that

PazYanoverr commented on docs/how-to/choose-configure-update-database.md:
```diff
@@ -0,0 +1,64 @@
+
+## Best Practices
+
+1. **Version Control**: Use Amplication's Git integration to track changes to your data model over time.
```
Not sure I understand this one, and how it's related to the database plugin...

dericksozo commented:
@PazYanoverr, the latest commit implements your feedback for this PR. PR #508 fixes a few things too so can it be merged first?

1. Regarding the broken develop your own custom plugins sidebar link, I fixed it in #508.
2. Regarding the plugin list, I updated it in #508 so we can keep it if that's alright with you.
3. In terms of fleshing out the authentication guide a bit more, I will update this one again in the next commit with more examples.

PazYanoverr commented on docs/getting-started/plugins.md:
```diff
@@ -1,43 +1,50 @@
-Plugins can be manually enabled or disabled on either of these pages.
-The source code of all community Amplication plugins are [available on GitHub](https://github.com/amplication/plugins).
+To see all published plugins, first visit the _Plugins_ tab in your service's dashboard and click on _All Plugins_ on the left sidebar.
+Your installed plugins are shown on the _Installed Plugins_ tab.
```
Still missing it

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -5,41 +5,41 @@ slug: /plugins/guides/authentication
-2. Choose and add an Auth Provider plugin (e.g., JWT, Auth0, etc.) that suits your needs.
-3. Configure the Authentication Entity in your service settings.
+2. Navigate to the "Authentication" category in the left sidebar.
+3. Choose and add an Auth Provider plugin (e.g., JWT, Auth0, etc.) that suits your needs.
```
Let's add here also ".NET" example, in addition to Auth0 etc.

PazYanoverr replied:
Again- let's refer to .NET option here (Supertokens is not available in .NET for now)

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -5,41 +5,41 @@ slug: /plugins/guides/authentication
-3. Configure the Authentication Entity in your service settings.
+2. Navigate to the "Authentication" category in the left sidebar.
+3. Choose and add an Auth Provider plugin (e.g., JWT, Auth0, etc.) that suits your needs.
+4. Install the "NestJS Auth Module" plugin if it's not already installed. It's a pre-requisite if you're adding Authentication to a Node.js service.
```
```suggestion
4. In case of adding Authentication to a Node.js service, make sure to Install the "NestJS Auth Module" plugin.
```

PazYanoverr replied:
I think this line- i.e. step # 4- should be before # 3 above.
In addition- do we need to add a note regarding the "User" entity? [The "User" entity is relevant for both .NET and Node.js? No so sure about it- can you check please?]

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -5,41 +5,41 @@ slug: /plugins/guides/authentication
 ### Node.js
 
-1. Auth0 Auth Provider
+1. [Auth0 Auth Provider](https://auth0.com/)
```
@dericksozo - Let's arrange it in the same order as we detail the plugins bloew

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -5,41 +5,41 @@ slug: /plugins/guides/authentication
-
-- For Node.js services, you can choose from the various authentication plugins mentioned above.
-- For .NET services, ASP.NET Core Identity is currently the primary authentication option. However, more authentication options for .NET will be added in the future.
+After adding the required plugins, you'll need to:
```
If we don't detail here how to do these, I guess we need to add links to the relevant documentation pages- 
how one can configure the auth settings in the plugin options? ("see below" is a valid options as well) 
how roles are defined?
etc.

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -81,3 +81,37 @@ If you're using a .NET service, refer to the [.NET Auth Core Identity plugin doc
+
+## Examples
+
+Here are some examples of how to use authentication in your service:
```
Can we come up with more examples? 
The page title is "auth plugin examples", and it should be some kind of a guide- can we update it to be like that, i.e. with more "real" examples?

PazYanoverr commented on docs/how-to/choose-configure-update-database.md:
```diff
@@ -8,27 +8,28 @@ slug: /plugins/guides/database
+1. When [creating a new service](/first-service/) using the wizard in Amplication, you'll be required to select a database. If you're using [Jovu](/amplication-ai/) to create your service, you can specify a different database than the default.
 2. Choose from the available options, which may include:
-   - PostgreSQL (default)
+   - PostgreSQL (default for Node.js)
```
Can you please what's the update for .NET as well?

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -7,39 +7,46 @@ pagination_next: getting-started/authentication
 ## Adding Authentication to Your Service
 
-If you initially created your service without authentication, you can still add it later. Here are the key steps:
+Before adding any specific authentication plugin, you must first enable the Authentication Entity in your service:
```
It's relevant only to Node.js, not to .NET.
In addition- it's not about "enable the Auth entitity"- but more about creating the authentication it, and defining it to be the auth entity in the service settings

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -7,39 +7,46 @@ pagination_next: getting-started/authentication
-4. Install the "NestJS Auth Module" plugin if it's not already installed. It's a pre-requisite if you're adding Authentication to a Node.js service.
-5. Configure the Authentication Entity in your service settings.
+1. Go to your service's Entities page.
+2. Locate or create an entity to serve as your Authentication Entity (commonly named "User").
```
The description is not correct. I think we need to say at the beginning that this is relevant only for services which were not initially created with auth. 
Then- we should should talk about creating the entity for the authentication. We should recommend creating a User entity, as we assist in creating all the required fields for such an entity

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -7,39 +7,46 @@ pagination_next: getting-started/authentication
-5. Configure the Authentication Entity in your service settings.
+1. Go to your service's Entities page.
+2. Locate or create an entity to serve as your Authentication Entity (commonly named "User").
+3. In the entity's settings, enable the "Authentication Entity" option.
```
We don't have such an option for an entity, but one needs to go to the Service settings (see below), and define the auth entity-
![image](https://github.com/user-attachments/assets/fe6b0f48-a165-4b8c-b25c-83149a9d19fe)

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -7,39 +7,46 @@ pagination_next: getting-started/authentication
+1. Navigate to your service's Plugins page.
+2. Go to the "Authentication" category in the left sidebar.
+3. For Node.js services, ensure you install the "NestJS Auth Module" plugin first.
+4. Choose and add an Auth Provider plugin that suits your needs (e.g., JWT, Auth0, Supertokens, etc.).
```
And for .NET ?

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -66,9 +73,20 @@ Amplication offers several authentication plugins to choose from. Each plugin ha
-### .NET 
+#### Basic Auth Provider
+
+- Enables a straightforward authentication scheme built into the HTTP protocol. - Requires sending user's credentials in the form of a username and password, encoded in base64, included in the Authorization header of the request.
```
```suggestion
- Enables a straightforward authentication scheme built into the HTTP protocol. Requires sending user's credentials in the form of a username and password, encoded in base64, included in the Authorization header of the request.
```

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -82,36 +100,148 @@ If you're using a .NET service, refer to the [.NET Auth Core Identity plugin doc
+2. Delete the User entity from your list of entities.
+3. Re-build your project and commit your changes to your preferred git provider.
+
 ## Examples
```
I'm not sure these are examples. It looks like we have here detailed explanations for some plugins' settings.
Is it correct? If so- let's update the title and the description below

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -82,36 +100,148 @@ If you're using a .NET service, refer to the [.NET Auth Core Identity plugin doc
 ## Examples
 
-Here are some examples of how to use authentication in your service:
+Let's look at detailed examples for some of the available authentication plugins:
```
Do we want to add some "real" examples like we had in this page before the changes?
Like how one can login using GraphQL, or REST API, and other examples (?).
@dericksozo - WDYT about it? Will it be useful?

dericksozo commented:
I appreciate the feedback, @PazYanoverr. I just implemented it.

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -11,27 +11,30 @@ Authentication and authorization are crucial components of your Amplication serv
+## Creating The Authentication Entity
 
-Before adding any specific authentication plugin, you must first enable the Authentication Entity in your service:
+If you initially created your service _without_ authentication, you must first create and define the Authentication Entity. The Authentication Entity **is required** to enable authentication on your service.
```
Again- it's relevant to Node.js and not .NET
I think we need to mention it and explain

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
```diff
@@ -245,3 +249,70 @@ The Keycloak Auth Provider integrates Keycloak authentication and authorization
+
+### Basic Authentication (Node.js)
+
+When using Basic HTTP authentication, when sending a request to the API you must provide a Basic HTTP authentication header with the format:
```
```suggestion
When using Basic HTTP authentication and sending a request to the API you must provide a Basic HTTP authentication header with the format:
```

PazYanoverr commented on docs/how-to/authentication-plugin-examples.md:
````diff
@@ -245,3 +249,70 @@ The Keycloak Auth Provider integrates Keycloak authentication and authorization
+Authorization: Basic YWRtaW46YWRtaW4=
+```
+
+By default, your app comes with one user with the username `admin` and password `admin`.
````
Is it relevant to the example?
I think this info is important and we should place it much higher in the doc and in a more dominant way

dericksozo commented:
@PazYanoverr, I implemented your feedback. I also added a new `<Tabs>` component to separate out the instructions for Node.js and .NET. We can start using this more throughout the docs. What are your thoughts?

PazYanoverr commented on docs/how-to/authentication-plugin-examples.mdx:
```diff
@@ -81,6 +101,10 @@ Amplication offers several authentication plugins to choose from. Each plugin ha
 - Requires sending user's credentials in the form of a username and password, encoded in base64, included in the Authorization header of the request.
 
+:::note
+If you use the Basic Auth Provider plugin, your service comes with one user with the username `admin` and the password `admin` by default.
```
Not sure, but I think that with each auth plugin, we supply the admin/admin user.
Can you please check for all the auth plugins including the .NET one?

PazYanoverr commented on docs/how-to/authentication-plugin-examples.mdx:
```diff
@@ -289,7 +313,7 @@ mutation {
+#### Header with JWT Included
 
 <!-- spell-checker: disable -->
 
```
The section is called "Real Examples" - let's omit the Real, as all the examples should be real...

PazYanoverr commented:
@dericksozo - Reviewed and added two more comments.
I liked the <Tabs> component, and, moving forward, we can certainly use it in more places as well. I found one issue with this component- it's not clear where the component "ends", i.e. which parts/section is influenced by the tab changes. Can it be handled easily?

PazYanoverr commented:
Thanks @dericksozo.
The new component looks great!

PazYanoverr commented:
@dericksozo -regarding the users-
The seed script of the generated code contains the first user details. The generated README file of the Node.js contains it as well-
.NET example (generated code)-
![image](https://github.com/user-attachments/assets/ca37f65d-e185-4c1b-943e-130ac2e63cf0)

Node.js example (generated code)-
![image](https://github.com/user-attachments/assets/44482166-d8f4-4228-8c29-1fcc6c1552a6)

Generated README file in Node.js-
![image](https://github.com/user-attachments/assets/5fddf5a5-d984-4a5a-86e2-fafaaa40bee2)

Generally speaking- we can say that for Node.js a default user of admin/admin is created, and in .NET - test@email.com / P@ssw0rd!

In addition- the above info appears in the plugin settings and the user can set it before generating the code.

Hope it makes sense

dericksozo commented:
The feedback is implemented for this one, @PazYanoverr. I cleaned it up a bit!
